src/Aspire.Dashboard/Otlp/Model/OtlpHelpers.cs

```diff
@@ -13,8 +14,27 @@
 
 namespace Aspire.Dashboard.Otlp.Model;
 
-public static class OtlpHelpers
+public static partial class OtlpHelpers
 {
+    private static readonly string s_longTimePatternWithMilliseconds = GetLongTimePatternWithMilliseconds();
+
+    static string GetLongTimePatternWithMilliseconds()
+    {
+        // From https://learn.microsoft.com/dotnet/standard/base-types/how-to-display-milliseconds-in-date-and-time-values
+
+        // Gets the long time pattern, which is something like "h:mm:ss tt" (en-US), "H:mm:ss" (ja-JP), "HH:mm:ss" (fr-FR).
+        var longTimePattern = DateTimeFormatInfo.CurrentInfo.LongTimePattern;
+
+        // Create a format similar to .fff but based on the current culture.
+        var millisecondFormat = $"{NumberFormatInfo.CurrentInfo.NumberDecimalSeparator}fff";
+
+        // Append millisecond pattern to current culture's long time pattern.
+        return MatchSecondsInTimeFormatPattern().Replace(longTimePattern, $"$1{millisecondFormat}");
+    }
+
+    [GeneratedRegex("(:ss|:s)")]
```

Comment:
Add a comment explaining the regex. Do some cultures only have `:s`?

Reply:
Do any cultures not use `:` as a delimiter? Is there a list containing possible `LongTimePattern` values that could be added to the comment here?

Reply:
I haven't observed that, though can't speak for all possible cultures. This approach comes from the MS docs, so at least had an SME review it.

If the match fails, then the default long time format comes through anyway, so the experience is not completely broken. You'd just miss millisecond resolution. But again, I can't see how that would happen.